src/pyodide/python-entrypoint-helper.js

```diff
@@ -98,42 +98,19 @@ const EMBEDDED_PYTHON_PACKAGES = [
 function transformMetadata(metadata) {
   // Workerd's metadata is slightly different. We transform it here to the format used upstream.
   if (metadata.globals !== undefined) {
-    return metadata;
+    return;
   }
-
-  var metadata = metadata;
-  metadata.globals = [];
-  for (const module of metadata.modules) {
-    if (metadata.mainModule === undefined) {
-      // The first module is the main module.
-      metadata.mainModule = module.name;
-    }
-
-    if (module.pythonModule !== undefined) {
-      metadata.globals.push({
-        name: module.name,
-        value: {
-          pythonModule: module.pythonModule,
-        },
-      });
-    }
-
-    if (module.pythonRequirement !== undefined) {
-      metadata.globals.push({
-        name: module.name,
-        value: {
-          pythonRequirement: module.pythonRequirement,
-        },
-      });
-    }
-  }
-  return metadata;
+  metadata.globals = metadata.modules.map((mod) => ({
```

Comment:
This looks cleaner than before, nice! One concern I had: Is it possible for modules to be metadata.modules to be empty? In that case accessing metadata.modules[0].name may cause issues.

Reply:
Thanks for the review @fhanau!

I think Workerd is required to have at least one module. Otherwise we get an error here:
https://github.com/cloudflare/workerd/blob/main/src/workerd/server/workerd-api.c++?plain=1#L208
Might be worth adding a comment though.